tests/run/py3k_super.pyx

```diff
@@ -45,6 +45,111 @@ class B(A):
             yield i
 
 
+class C(A):
+    """
+    >>> obj = C()
+    >>> obj.method_1()
+    2
+    >>> obj.method_2()
+    3
+    >>> obj.method_3()
+    ['__class__', 'self']
```

Comment:
And is `__class__` really `C` here (or `A` by accident)?

Reply:
Good point; I've added tests for this below.